### PR TITLE
fix: missing cloud profile network isolation

### PR DESCRIPTION
**Pull request by vknabel**
As discussed, this adds the missing rendering of the network isolation

### CODE REVIEW

File: control-plane/roles/gardener/templates/30-cloud-profile-metal.yaml [hunk 1]
```diff
@@ -30,6 +30,9 @@ spec:
 {% for id, partition_config in gardener_cloud_profile_partitions.items() %}
           {{ id }}:
             firewallTypes: {{ partition_config.get('default-machine-types', {}).get('firewall', []) | to_json }}
+{% if 'network-isolation' in partition_config.get('default-machine-types', {}) %}
+            networkIsolation: {{ partition_config.get('default-machine-types', {}).get('network-isolation', []) | to_json }}
+{% endif %}
```
Comment (robertvolkmann) [inline control-plane/roles/gardener/templates/30-cloud-profile-metal.yaml:35]:
```suggestion
{% if 'network-isolation' in partition_config %}
            networkIsolation: |
              {{ partition_config.get('network-isolation') | to_yaml | indent(width=14, first=false) }}
{% endif %}
```

File: control-plane/roles/gardener/templates/30-cloud-profile-metal.yaml [hunk 2]
```diff
@@ -30,6 +30,10 @@ spec:
 {% for id, partition_config in gardener_cloud_profile_partitions.items() %}
           {{ id }}:
             firewallTypes: {{ partition_config.get('default-machine-types', {}).get('firewall', []) | to_json }}
+{% if 'network-isolation' in partition_config %}
+            networkIsolation: |
+              {{ partition_config.get('network-isolation') | to_yaml | indent(width=14, first=false) }}
```
Comment (robertvolkmann) [inline control-plane/roles/gardener/templates/30-cloud-profile-metal.yaml:35]: Sry, we use in the whole file to_Json.
```suggestion
            networkIsolation: {{ partition_config.get('network-isolation') | to_json }}
```

**Review by robertvolkmann — APPROVED** [review]

**Comment by majst01** [comment]
Has this been deployed already in the test environment ?

**Comment by vknabel** [comment]
@majst01 this was copied from our overriding template in the deployment, but we forgot to migrate this to the metal-roles back then. The override has definitely been deployed in test.

**Review by majst01 — APPROVED** [review]

**Pull request merged**